### DISCUSSION TITLE
Hosting Lowdefy as a static site

### DISCUSSION THREAD

**Discussion by em3ndez** [lowdefy/lowdefy#537] [Ideas and feature requests]
I'd like to know if it is possible to deploy a lowdefy application in an Apache webserver. What I mean is if it's possible to run lowdefy inside the javascript engine of a browser (perhaps with certain limitations) instead of inside a Node engine.

**Comment by SamTolmay** [comment]
Hi @em3ndez 

To confirm we are understanding correctly, you would like to host a Lowdefy app as static content, served by an Apache webserver?

We have been thinking about doing this, it would be possible to adapt Lowdefy to build as a static site, but it isn't possible right now.

On a static version of Lowdefy, you wouldn't have any user authentication or authorization, and you wouldn't have any connections/requests/secrets. We would implement a client fetch action, so you would be able to make HTTP requests from the client. Most other things should still be possible.

This is something we would like to build and can prioritise if we have enough requests for this.

I'm converting this to a discussion so we can discuss it there further.

What is your use case for this, and what features would be important to you?

**Comment by SamTolmay** [comment]
Changed the title to improve discoverabilty

**Comment by em3ndez** [comment]
> You would like to host a Lowdefy app as **static content**, served by an Apache webserver?
- Correct.

> On a static version of Lowdefy, you wouldn't have any user authentication or authorization
- That's right. The authentication/authorization has to be already done in some other way.

> We would implement a client fetch action, so you would be able to make HTTP requests from the client.
- That's the idea: to have a REST client capable of invoking things from a server.

> What is your use case for this, and what features would be important to you?
- The use case would be to have an easy way to add functionality to an already deployed Java application without increasing more the infrastructure. It is true that it would have some limitations but, with what you have already, there is a lot of common ground covered. Besides, if someone needs more, you already have other deployments targets more suitable to those. Regarding features I think that's all for now: a way for calling REST endpoints (I think you got something done already) and a Webpack configuration that packs it all for a static deployment.

**Comment by vaddisrinivas** [comment]
I totally second this feature request.  

Even from my perspective, a static site that is rendered/compiled beforehand and served via any web server would really be very useful.
While user interaction is important for many, there can be use-cases like a wiki/ docs, announcements pages, simple blogs, status page e.t.c. where the user interaction could be minimal.

Using Lowdefy to generate UI  via YAML and being able to serve such static content at a blazing pace directly via webservers/CDN's(caching)  could be really huge!
